### DISCUSSION TITLE
Similarity between MCUT and Manifold?

### DISCUSSION THREAD

**Discussion by chitalu** [elalish/manifold#188] [General]
Hi,

Cool library. 

Can you provide some insight as to how Manifold differs from [MCUT](https://github.com/cutdigital/mcut), which is a library I develop? 

best,
Chitalu

**Comment by elalish** [comment]
Likewise! Thanks for reaching out. They do look broadly similar in their goals, though it appears yours operates on both manifold and open meshes (mine does only manifold input/output). You also appear to use an exact geometry kernel, which is the method most good computational geometry uses. Mine has a different underlying system which embraces floating-point error, but guarantees manifoldness even for highly degenerate situations. Really that's all that sets it apart from what's come before. 

I'd welcome comparisons though; I'm sure you've found some difficult test cases - I'd love to know if you can find problems with mine. Also, you mention using high-performance parallel computing; does it take advantage of the GPU? I'm making as much of mine CUDA-capable as possible, which is giving it decent performance (it's improved a bit from [here](https://elalish.blogspot.com/2022/03/manifold-performance.html)). How does yours compare to CGAL? That's been my benchmark so far.

**Reply by chitalu** [reply]
Thanks. I'll test out Manifold in due course.

The is no GPU parallelism in MCUT at the moment (only CPU-based multi-threading ). However, this will be added in the long term. OpenCL will probably be the preferred option for portability. 

I am yet to make extensive performance measurements but MCUT should be on-par with [Carve](https://github.com/VTREEM/Carve/blob/master/lib/carve.cpp). I'll publish results after next release, which will also include comparisons against CGAL's corefinement routines.

Cheers

**Comment by fire** [comment]
The license of MCUT (GPL) is not compatible with use in commercial products such as commercial games using Godot Engine.

**Reply by chitalu** [reply]
Hi -

MCUT is actually available under an Open Source license as well as a commercial license. Users choosing to use MCUT under the free-of-charge Open Source license have to comply to its terms, otherwise they have to buy a commercial license.

You can check [here](https://cutdigital.github.io/mcut.site/license/) for more info.
